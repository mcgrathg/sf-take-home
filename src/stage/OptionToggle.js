import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import colors from '../config/colors';

const StyledToggle = styled.button`
  max-width: 11rem;
  margin: 0 0.25rem 0.75rem;
  padding: 0.5rem 1rem;
  border: 1px solid ${colors.greyBorder};
  border-radius: 1.5rem;
  background-color: ${colors.white};

  @media (min-width: 500px) {
    margin: 0 0.5rem 1rem;
    padding: 0.5rem 2rem;
  }

  &:focus,
  &.checked:focus:active {
    box-shadow: 0 0 0 2px ${colors.greyLight}, 0 0 0 3px ${colors.greyBorder};
  }

  &:active:focus,
  &.checked:focus {
    box-shadow: 0 0 0 2px ${colors.greyLight}, 0 0 0 3px ${colors.turquoise};
  }

  &:active {
    transform: scale(1.1);
    background-color: ${colors.turquoise};
    color: ${colors.white};
  }

  &.checked {
    border-color: ${colors.turquoise};
    color: ${colors.white};
    background-color: ${colors.turquoise};

    &:active {
      background-color: ${colors.white};
      color: inherit;
      border-color: ${colors.greyBorder};
    }

    &.show-off {
      transform: scale(1.05);
    }

    &.downplay {
      transform: scale(0.95);
    }
  }
`;

const OptionToggle = ({ option, isSelected, onToggled, className }) => (
  <StyledToggle
    key={option}
    type="button"
    aria-pressed={isSelected ? 'true' : 'false'}
    className={`option ${className} ${isSelected ? 'checked' : ''}`}
    onClick={() => {
      onToggled(option, !isSelected);
    }}
  >
    {option}
  </StyledToggle>
);

OptionToggle.propTypes = {
  option: PropTypes.string.isRequired,
  isSelected: PropTypes.bool.isRequired,
  onToggled: PropTypes.func.isRequired,
  className: PropTypes.string,
};

OptionToggle.defaultProps = {
  className: '',
};

export default OptionToggle;
