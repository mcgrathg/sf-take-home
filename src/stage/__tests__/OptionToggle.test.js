import React from 'react';
import { render } from 'react-testing-library';
import { noOp } from 'test-utils';
import OptionToggle from '../OptionToggle';

test('renders', () => {
  const { container } = render(
    <OptionToggle option="" isSelected={false} onToggled={noOp} />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
