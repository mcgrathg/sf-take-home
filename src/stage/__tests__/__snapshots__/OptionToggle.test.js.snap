// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  max-width: 11rem;
  margin: 0 0.25rem 0.75rem;
  border: 1px solid #c0bdba;
  border-radius: 1.5rem;
  padding: 0.5rem 1rem;
  background-color: #fff;
}

.c0:focus,
.c0.checked:focus:active {
  box-shadow: 0 0 0 2px #f2f0ee,0 0 0 3px #c0bdba;
}

.c0:active:focus,
.c0.checked:focus {
  box-shadow: 0 0 0 2px #f2f0ee,0 0 0 3px $turquoise;
}

.c0:active {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  background-color: #1695a3;
  color: #fff;
}

.c0.checked {
  border-color: #1695a3;
  color: #fff;
  background-color: #1695a3;
}

.c0.checked:active {
  background-color: #fff;
  color: inherit;
  border-color: #c0bdba;
}

@media (min-width:500px) {
  .c0 {
    margin: 0 0.5rem 1rem;
    padding: 0.5rem 2rem;
  }
}

<button
  aria-pressed="false"
  class="option  c0"
  type="button"
/>
`;
